lib/msal-node/src/network/HttpClientWithRetries.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    HeaderNames,
+    INetworkModule,
+    NetworkRequestOptions,
+    NetworkResponse,
+} from "@azure/msal-common";
+import { IHttpRetryPolicy } from "../retry/IHttpRetryPolicy";
+import http from "http";
+import { HttpMethod } from "../utils/Constants";
+
+export class HttpClientWithRetries implements INetworkModule {
+    private httpClientNoRetries: INetworkModule;
+    private retryPolicy: IHttpRetryPolicy;
+
+    constructor(
+        httpClientNoRetries: INetworkModule,
+        retryPolicy: IHttpRetryPolicy
+    ) {
+        this.httpClientNoRetries = httpClientNoRetries;
+        this.retryPolicy = retryPolicy;
+    }
+
+    private retryAfterMillisecondsToSleep(
```

Comment:
In case we use the retry header to make retries in MSAL what will happen if there are timeouts setup? There are cancellation tokens in dotnet, can user set a timeout for an MSAL request in JS?

Reply:
Yes, that functionality already exists in msal-node: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/src/network/HttpClient.ts#L67

Reply:
currently the cancellationToken functionality only exists for post requests

Reply:
I'm going to make another GitHub issue for this, because there appears to be some inconsistencies in the msal-node code.

Reply:
Should we honor the retry header in this PR? Since managed identity responses does not contain a retry header, these are more relevant to calls to ests. Also if we implement the retry header we will need to also have the logic to honor the timeout/ cancellation tokens. IMO break the PR to have the logic for retry-after header with ests requests. 
Thoughts @bgavrilMS

Reply:
It's fine, @Robbie-Microsoft already implemented and unit test this ( I agree it should have been a different PR). 

We discussed that all retry policies should obey Retry-After header.